content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -290,11 +304,11 @@ class CreateAndIssueMarsVoucherInitiator @JsonConstructor constructor(private va
 
 ### Write the responder flow
 
-Now that you've written the initiating flow, write the responder flow. The responder flow in the `CreateAndIssueMarsVoucher` flow is very simple, as you don't need to verify anything.
+Now that you've written the initiating flow, write the responder flow that responds to the request to update the ledger. The responder flow in the `CreateAndIssueMarsVoucher` flow is very simple and in the context of the Mission Mars CorDapp, you don't need to verify anything. However, when signing a responder flow in a "real-life context", do not sign it before verifying its contents and ensuring you are accepting what you want to accept.
 
 1. Add the `@InitiatedBy` annotation. This indicates that this is the responder flow.
 
-2. Add in an empty responder flow:
+2. Add in a responder flow:
```

Comment:
@peterli-r3 Could you please give a description of what's happening in the responder flow?

Reply:
When the responder receives a half signed transaction from an initiating party, the responder flow is most likely being asked to sign the transaction. So one thing the responder flow will do is to go through some checks on the transaction (implemented in the responder flow), and if all the checks are passed, it will sign the transaction and return back to the initiating party. The second thing a responder will do is to receive and record a fully signed transaction to its local ledger (database). This is done by the ReceiveFinalityFlow subflow.

Reply:
added